lib/grpc_generator.dart

```diff
@@ -0,0 +1,252 @@
+
+  void generate(IndentingWriter out) {
+    _generateClient(out);
+    out.println();
+    _generateService(out);
+  }
+
+  void _generateClient(IndentingWriter out) {
+    out.addBlock('class $_clientClassname {', '}', () {
+      out.println('final ClientChannel _channel;');
+      out.println();
+      for (final method in _methods) {
+        method.generateClientMethodDescriptor(out);
+      }
+      out.println();
+      out.println('$_clientClassname(this._channel);');
+      for (final method in _methods) {
+        method.generateClientStub(out);
+      }
+    });
+  }
+
+  void _generateService(IndentingWriter out) {
+    out.addBlock('abstract class $_serviceClassname extends Service {', '}',
+        () {
+      out.println('String get \$name => \'$_fullServiceName\';');
+      out.println();
+      out.addBlock('$_serviceClassname() {', '}', () {
+        for (final method in _methods) {
+          method.generateServiceMethodRegistration(out);
+        }
+      });
+      out.println();
+      for (final method in _methods) {
+        method.generateServiceMethodPreamble(out);
+      }
+      for (final method in _methods) {
+        method.generateServiceMethodStub(out);
+      }
+    });
+  }
+}
+
+class _GrpcMethod {
+  final MethodDescriptorProto _method;
+
+  final GrpcServiceGenerator _service;
+
+  String _grpcName;
+  String _dartName;
+  String _serviceName;
+
+  bool _clientStreaming;
+  bool _serverStreaming;
+
+  String _requestType;
+  String _responseType;
+
+  String _argumentType;
+  String _clientReturnType;
+  String _serverReturnType;
+
+  _GrpcMethod(this._service, this._method) {
+    _grpcName = _method.name;
+    _dartName =
+        _grpcName.substring(0, 1).toLowerCase() + _grpcName.substring(1);
+    _serviceName = _service._fullServiceName;
+
+    _clientStreaming = _method.clientStreaming;
+    _serverStreaming = _method.serverStreaming;
+  }
+
+  void resolve(GenerationContext ctx) {
+    _service._addDependency(ctx, _method.inputType, "input type of $_grpcName");
+    _service._addDependency(
+        ctx, _method.outputType, "output type of $_grpcName");
+
+    _requestType = _service._getDartClassName(_method.inputType);
+    _responseType = _service._getDartClassName(_method.outputType);
+
+    _argumentType = _clientStreaming ? 'Stream<$_requestType>' : _requestType;
+    _clientReturnType = _serverStreaming
+        ? 'ResponseStream<$_responseType>'
+        : 'ResponseFuture<$_responseType>';
+    _serverReturnType =
+        _serverStreaming ? 'Stream<$_responseType>' : 'Future<$_responseType>';
+  }
+
+  void generateClientMethodDescriptor(IndentingWriter out) {
+    out.println(
+        'static final _\$$_dartName = new ClientMethod<$_requestType, $_responseType>(');
+    out.println('    \'/$_serviceName/$_grpcName\',');
+    out.println('    ($_requestType value) => value.writeToBuffer(),');
+    out.println(
+        '    (List<int> value) => new $_responseType.fromBuffer(value));');
```

Comment:
This is a place where the indentation kinda hurts the readability (and it is painful to maintain as you change the code). Does it matter now that you use the formatter?